### ISSUE TITLE
Deleting one out of two ParameterCollection

### ISSUE THREAD

**Issue by smar111** [clab/dynet#1193]
Hi,

So I have two ParameterCollections in my architecture. One of them is just used to initialize some values of the other. I was wondering if I can delete it after I have done that for better memory optimization?

**Comment by neubig** [comment]
I'm not sure exactly what you're trying to do. Do you have a pre-trained model that you want to use to initialize some subset of the parameters of another model? If so, you can do this by using the `load()` and `populate()` commands from the DyNet model reader, passing in the name of the parameters you want to load in. This will allow you to not create a separate `ParameterCollection`.

You can find an example in the unit tests here: https://github.com/clab/dynet/blob/master/tests/test-io.cc#L86

But we need better examples, I'll open a separate issue about needing better examples of IO.

**Comment by smar111** [comment]
Yes but I was wondering if it works for parameters of type "Builder", as it is not given in the example.

Let me explain what I am trying to do. I have a pre-trained model, say R, containing an LSTM. I have another model M, with separate parameters than those in R. I have to run R and get some representations which are needed by the model M. Once I get those, I don't need R anymore and I was thinking it best to somehow delete it's parameters from memory.
